### PR TITLE
Remove background tasks on sign out

### PR DESCRIPTION
**Pull request by TomFanella4**
## Changes
1. Unregister Location listener
1. Unregister Notification listener
1. Sign out of Firebase

### CODE REVIEW

File: client/app/navigation/MainTabNavigator.js [hunk 1]
```diff
@@ -87,30 +89,38 @@ class MainTabNavigation extends React.Component {
...
   componentDidMount() {
-    this._registerForLocationUpdating();
-    this._registerForPushNotifications();
-    this.props.registerFetchThreadListAsync();
+    Promise.all([
+      this._registerForLocationUpdating(),
+      this._registerForPushNotifications(),
+      this.props.registerFetchThreadListAsync()
+    ])
+    .then(values => {
```
Comment (mpace965) [inline client/app/navigation/MainTabNavigator.js]: You can pattern match against arrays, eg:

```js
.then(([locationSub, notificationSub]) => {
// ...
})
```

might be more clear. Not really necessary, but seeing this made me remember that you can pattern match arrays in JavaScript.

**Review by mpace965 — APPROVED** [review]
Code looks good, and I believe it works for me. What is the recommended method of testing here?

File: client/app/navigation/MainTabNavigator.js (hunk 1 above)
Comment (TomFanella4) [reply client/app/navigation/MainTabNavigator.js]: How to test:
1. Sign out
2. You shouldn't receive an error

Comment (mpace965) [reply client/app/navigation/MainTabNavigator.js]: All is good then

**Pull request merged**